zilliqa/src/exec.rs

```diff
@@ -506,6 +502,20 @@ impl State {
         Ok((e, state.finalize(), cfg.env))
     }
 
+    // The rules here are somewhat odd, and inherited from ZQ1
```

Comment:
@bzawisto Could you double-check these rules are consistent with how we figured Ethereum transactions charge gas when the gas limit is reached?